tests/units/test_openstackmap.py

```diff
@@ -0,0 +1,210 @@
+            resp.status_code = OK
+            resp._content = json_data
+
+        elif url == '/role_assignments':
+            json_data = open(UNIT_TEST_RESOURCES_FOLDER + LIST_ROLE_ASSIGNMENTS_RESPONSE_FILE).read()
+            resp.status_code = OK
+            resp._content = json_data
+
+        return resp
+
+
+class TestOpenstackMap(TestCase):
+
+    mock_session = MySessionMock()
+
+    def setUp(self):
+        self.OS_AUTH_URL = 'http://cloud.host.fi-ware.org:4731/v2.0'
+        self.OS_USERNAME = 'user'
+        self.OS_PASSWORD = 'password'
+        self.OS_TENANT_NAME = 'user cloud'
+        self.OS_TENANT_ID = '00000000000000000000000000000001'
+        self.OS_REGION_NAME = 'Spain2'
+        self.OS_TRUST_ID = ''
+        self.OS_KEYSTONE_ADMIN_ENDPOINT = 'http://cloud.lab.fiware.org:4730'
+
+        environ.setdefault('OS_AUTH_URL', self.OS_AUTH_URL)
+        environ.setdefault('OS_USERNAME', self.OS_USERNAME)
+        environ.setdefault('OS_PASSWORD', self.OS_PASSWORD)
+        environ.setdefault('OS_TENANT_NAME', self.OS_TENANT_NAME)
+        environ.setdefault('OS_TENANT_ID', self.OS_TENANT_ID)
+        environ.setdefault('OS_REGION_NAME', self.OS_REGION_NAME)
+        environ.setdefault('OS_TRUST_ID', self.OS_TRUST_ID)
+
+    @patch.object(os, 'mkdir')
+    @patch.object(os.path, 'exists')
+    def test_implement_openstackmap(self, mock_exists, mock_os):
+        """test_implement_openstackmap check that we could build an empty map from the resources (VMs, networks, images,
+        volumes, users, tenants, roles...) in an OpenStack infrastructure."""
+        mock_exists.return_value = False
+        mock_os.return_value = True
+        openstackmap = OpenStackMap(region=self.OS_REGION_NAME, auto_load=False)
+        self.assertTrue(mock_os.called)
+        self.assertIsNotNone(openstackmap)
+
+    def test_implement_openstackmap_without_region_name(self):
+        """test_implement_openstackmap_without_region_name check that we could not build an empty map without providing
+        a region Name."""
+
+        del os.environ['OS_REGION_NAME']
+
+        with self.assertRaises(Exception):
+            OpenStackMap(auth_url=self.OS_AUTH_URL, auto_load=False)
+
+    @patch('utils.osclients.session', mock_session)
+    @patch.object(os, 'mkdir')
+    @patch.object(os.path, 'exists')
+    def test_implement_openstackmap_with_keystone_admin_endpoint(self, mock_exists, mock_os):
+        """test_implement_openstackmap_with_keystone_admin_endpoint check that we could build an empty map from the
+         resources providing a keystone admin endpoint environment variable."""
+
+        environ.setdefault('KEYSTONE_ADMIN_ENDPOINT', self.OS_AUTH_URL)
+
+        mock_exists.return_value = False
+        mock_os.return_value = True
+        openstackmap = OpenStackMap(auto_load=False)
+        self.assertTrue(mock_os.called)
+        self.assertIsNotNone(openstackmap)
+        del os.environ['KEYSTONE_ADMIN_ENDPOINT']
```

Comment:
I don't think so, because there are different test cases and that sentence is not used in all methods.